pwndbg/checksec.py

```diff
@@ -8,53 +8,52 @@
 from __future__ import print_function
 from __future__ import unicode_literals
 
-import sys
 import subprocess
 
 import pwndbg.file
 
-def checksec():
-    local_path = pwndbg.file.get_file(pwndbg.proc.exe)
-    result = {}
-
+def checksec(path):
     for program in ['checksec', 'checksec.sh']:
         program = pwndbg.which.which(program)
         if program:
-            cs_out = subprocess.check_output([program, '--file', local_path]).decode('utf-8')
-
-            if "Full RELRO" in cs_out:
-                result['RELRO'] = 2
-            if "Partial RELRO" in cs_out:
-                result['RELRO'] = 1
-            if "No RELRO" in cs_out:
-                result['RELRO'] = 0
-
-            if "Canary found" in cs_out:
-                result['CANARY'] = 1
-            else:
-                result['CANARY'] = 0
-
-            if "NX enabled" in cs_out:
-                result['NX'] = 1
-            else:
-                result['NX'] = 0
-
-            if "PIE enabled" in cs_out:
-                result['PIE'] = 1
-            else:
-                result['PIE'] = 0
-
-            if "No RPATH" in cs_out:
-                result['RPATH'] = 0
-            else:
-                result['RPATH'] = 1
-
-            if "No RUNPATH" in cs_out:
-                result['RUNPATH'] = 0
-            else:
-                result['RUNPATH'] = 1
-
-            return result
+            result = {}
+            try:
+                cs_out = subprocess.check_output([program, '--file', path]).decode('utf-8')
+                if "Full RELRO" in cs_out:
+                    result['RELRO'] = 2
+                if "Partial RELRO" in cs_out:
+                    result['RELRO'] = 1
+                if "No RELRO" in cs_out:
+                    result['RELRO'] = 0
+
+                if "Canary found" in cs_out:
+                    result['CANARY'] = True
+                else:
+                    result['CANARY'] = False
+
+                if "NX enabled" in cs_out:
+                    result['NX'] = True
+                else:
+                    result['NX'] = False
+
+                if "PIE enabled" in cs_out:
+                    result['PIE'] = True
+                else:
+                    result['PIE'] = False
+
+                if "No RPATH" in cs_out:
+                    result['RPATH'] = False
+                else:
+                    result['RPATH'] = True
+
+                if "No RUNPATH" in cs_out:
+                    result['RUNPATH'] = False
+                else:
+                    result['RUNPATH'] = True
+                return result
+
+            except:
```

Comment:
This try/except handles too much code and too broad exception - imagine if someone modifies e.g. line 30 and makes some stupid mistake - he/she would waste time looking for that :p.

Just do:
```python
try:
    cs_out = subprocess.check_output([program, '--file', path]).decode('utf-8')
except (OSError, subprocess.CalledProcessError):
    ...
```

(like here https://github.com/pwndbg/pwndbg/blob/stable/pwndbg/version.py#L24-L27 - however now I think this tries too much code too, huh)